### PR TITLE
Topic/training step tracking

### PR DESCRIPTION
**Pull request by steventlamb**

### CODE REVIEW

**Comment by steventlamb** [comment]
Reworked in response to this comment from @jwalgran 

> Parks/TreeKit is working on the structure of the training materials. Right now, they are proposing 4 major sections, which would be shown as checklist items on the dashboard:

> Getting Started
The Mapping Method
Tree Data
Tree Surrounding
(reference: http://sgowbu.axshare.com/#p=treekit_2015_census_sitemap)

> We should use those 4 as our boolean columns: visited_training_section_*

Ready for review.

**Comment by maurizi** [comment]
@azavea-bot rebuild

File: src/nyc_trees/apps/home/views.py [hunk 1]
```diff
@@ -21,6 +24,28 @@ def retrieve_job_status(request):
...
+            ctx = view_fn(request, *args, **kwargs)
+            user = request.user
+            if user.is_authenticated() and not getattr(user, attr):
+                setattr(user, attr, True)
+                user.save()
+            return ctx
+        return wrapper
+    return outer_decorator
+
+
```
Comment (maurizi) [inline src/nyc_trees/apps/home/views.py]: Neither of these functions are really views.  Consider moving both of these into a new `home/decorators.py` file.

File: src/nyc_trees/apps/home/views.py [hunk 2]
```diff
@@ -21,6 +24,28 @@ def retrieve_job_status(request):
     pass
 
 
+def redirect_to_flat_page(url):
```
Comment (kdeloach) [inline src/nyc_trees/apps/home/views.py]: Does this actually redirect to a flatpage? Or does it just return the context?

File: src/nyc_trees/apps/home/views.py [hunk 3]
```diff
@@ -21,6 +24,28 @@ def retrieve_job_status(request):
...
+
+def mark_user_on_success(attr):
+    def outer_decorator(view_fn):
+        @wraps(view_fn)
+        def wrapper(request, *args, **kwargs):
+            # do this first to make sure the wrapped view doesn't
+            # raise
+            ctx = view_fn(request, *args, **kwargs)
+            user = request.user
+            if user.is_authenticated() and not getattr(user, attr):
```
Comment (kdeloach) [inline src/nyc_trees/apps/home/views.py]: Consider adding statsd tracking here.

File: src/nyc_trees/apps/home/views.py (hunk 2 above)
Comment (steventlamb) [inline src/nyc_trees/apps/home/views.py]: redirect is probably the wrong word. It does not perform an HttpResponseRedirect. It renders a flat page. Not a context, it returns rendered markup.

**Comment by kdeloach** [comment]
+1

File: src/nyc_trees/apps/home/views.py (hunk 2 above)
Comment (steventlamb) [inline src/nyc_trees/apps/home/views.py]: The function that this calls is how flatpage url/view routing works under the hood.

**Comment by steventlamb** [comment]
This underwent a fair amount of refactoring when new features were added. Most of the comments on this PR were addressed during some rewrites and refactors in #345. They'll be included in that PR, or documented when not yet completed. Closing this PR as these commits are on that PR too.

**Pull request closed without merging**